apps/emqx_resource/src/emqx_resource_worker.erl

```diff
@@ -425,18 +376,23 @@ flush(Data0) ->
         batch_size := BatchSize,
         queue := Q0
     } = Data0,
-    case replayq:count(Q0) of
+    Data1 = cancel_flush_timer(Data0),
+    case queue_count(Q0) of
         0 ->
-            Data = cancel_flush_timer(Data0),
-            {keep_state, Data};
+            {keep_state, Data1};
         _ ->
```

Comment:
maybe that could get the buffer worker stuck?  https://github.com/emqx/emqx/pull/9699#discussion_r1069457029

Reply:
I think `resume` only makes sense for the blocked state?  Since it's to retry the inflight table when blocked for a while, or when a response is received and frees some space in the inflight table.